app/src/main/java/com/android/unio/ui/user/UserProfileEdition.kt

```diff
@@ -391,3 +449,38 @@ private fun SocialButtonAndFlowRow(
     }
   }
 }
+
+@Composable
+fun UserDeletePrompt(
+    onDismiss: () -> Unit,
+    onConfirmDelete: () -> Unit,
+) {
+  val context = LocalContext.current
+
+  Dialog(
+      onDismissRequest = onDismiss,
+      properties = DialogProperties(usePlatformDefaultWidth = false)) {
+        Card(
+            elevation = CardDefaults.cardElevation(8.dp),
+            shape = RoundedCornerShape(16.dp),
+            modifier = Modifier.fillMaxWidth().padding(20.dp).testTag("")) {
+              Column(
+                  modifier = Modifier.padding(16.dp),
+                  horizontalAlignment = Alignment.CenterHorizontally,
+                  verticalArrangement = Arrangement.SpaceBetween) {
+                    Text(
+                        text = context.getString(R.string.user_edition_delete_user_confirmation),
+                        textAlign = TextAlign.Center)
+
+                    Button(
+                        modifier = Modifier.testTag(UserEditionTestTags.DELETE_CONFIRMATION),
+                        onClick = onConfirmDelete,
```

Comment:
You do not check anywhere whether the user has an Internet connection. As mentioned above, this causes problems in offline mode. The solution is quite simple, as it's understandable to deny the ability to delete the user in offline mode. See line 232 how this is done.